core/lib/zksync_core/src/metadata_calculator/mod.rs

```diff
@@ -184,87 +176,4 @@ impl MetadataCalculator {
             .loop_updating_tree(self.delayer, &pool, stop_receiver, self.health_updater)
             .await
     }
-
-    /// This is used to improve L1 gas estimation for the commit operation. The estimations are computed
-    /// in the State Keeper, where storage writes aren't yet deduplicated, whereas L1 batch metadata
-    /// contains deduplicated storage writes.
-    async fn reestimate_l1_batch_commit_gas(
-        storage: &mut StorageProcessor<'_>,
-        header: &L1BatchHeader,
-        metadata: &L1BatchMetadata,
-    ) {
-        let estimate_latency = METRICS.start_stage(TreeUpdateStage::ReestimateGasCost);
-        let unsorted_factory_deps = storage
-            .blocks_dal()
-            .get_l1_batch_factory_deps(header.number)
-            .await
-            .unwrap();
-        let commit_gas_cost =
-            commit_gas_count_for_l1_batch(header, &unsorted_factory_deps, metadata);
-        storage
-            .blocks_dal()
-            .update_predicted_l1_batch_commit_gas(header.number, commit_gas_cost)
```

Comment:
Dumb question: Is this update no longer required? If so, does it make sense to remove `update_predicted_l1_batch_commit_gas()` from DAL?